changelog/dmd.traits_getOverloads_with_symbol.dd

```diff
@@ -0,0 +1,14 @@
+__traits(getOverloads) takes an overload symbol
+
+`__traits(getOverloads)` can now be invoked with
+a single parameter. This parameter must be an overload set.
```

Comment:
I'm confused about what this means. Each overload of foo in a different module forms a different overload set:
```
module a;
void foo();
void foo(int);
```
module b;
void foo();
```
There are two overload sets here, the one for `a` and one for `b`. How is this going to work with this change?

Reply:
Overload sets can also have as members other overload sets.

I can't tell from the bug report, the documentation, or the comments, what this is going to do.

Reply:
If you're in a module where you see both `a` and `b`, and you specify `foo`, you consider all three as valid overloads. It uses the lexical scope of the use-site.

Reply:
An overload _set_ is something specific in D, and is not just an aggregation of all the instances of a name that is in scope. I.e. it is not a tuple.

It cannot behave the same, as `templ!foo` will give an error if more than one `foo` exists.

Reply:
I'm confused:

```
void foo(int a) {}
void foo() {}
void bar(alias x)() { x(5); x(); }

void main() {
    bar!foo();
}
```

This also works if you split `foo` into two separate modules.